backend/lib/services/terminals/terminalBootstrap.js

```diff
@@ -53,39 +54,70 @@ const { getSeed } = require('../../cache')
 
 const TERMINAL_KUBE_APISERVER = 'dashboard-terminal-kube-apiserver'
 
-class BootstrapPendingSet extends Set {
-  _keyForResource (resource) {
-    const kind = resource.kind
-    const { metadata: { name, namespace } } = resource
-    return `${kind}/${namespace}/${name}`
+class BootstrapSet extends Set {
+  static keyForResource () {
+    throw new Error('needs to be overwritten in subclass')
   }
 
   containsResource (resource) {
-    const key = this._keyForResource(resource)
+    const key = this.constructor.keyForResource(resource)
     return this.has(key)
   }
 
   removeResource (resource) {
-    const key = this._keyForResource(resource)
+    const key = this.constructor.keyForResource(resource)
     return this.delete(key)
   }
 
   addResource (resource) {
-    const key = this._keyForResource(resource)
+    const key = this.constructor.keyForResource(resource)
+    return this.addResourceWithKey(key)
+  }
+
+  addResourceWithKey (key) {
     this.add(key)
     return key
   }
 }
 
+class BootstrapPendingSet extends BootstrapSet {
+  static keyForResource (resource) {
+    const kind = resource.kind
+    const { metadata: { name, namespace } } = resource
+    return `${kind}/${namespace}/${name}`
+  }
+}
+
+class BootstrappedSet extends BootstrapSet {
+  static keyForResource (resource) {
+    const { metadata: { uid } } = resource
+    return uid
+  }
+}
+
+function taskIdForResource (resource) {
+  const { metadata: { uid } } = resource
+  return uid
+}
+
 class Handler {
-  constructor (fn, description) {
+  constructor ({ id, fn, description }) {
+    this.id = id
     this.fn = fn
-    this._run = () => fn()
+    this._run = session => fn(session)
```

Comment:
Why do we store the session on the handler if we pass it to the run method. Or the other way around. Why do we pass the session to the run method if we store it on the handler?

Reply:
so that we can cancel it later on?